AWS EC2 tweaks for Debian images

msheiny opened the pull request:
Nothing crazy, just noticed that the submenu text was coming up different and I
added a config that I was able to compile against. Note the `awsfull` designator
thats because this config includes all the bloat that I considered included with
the default AMI config.

msheiny committed:
AWS AMI Debian fix for grub submenu text

Discovered on the default AWS Debian AMIs (owner `379101102735`)
that GRUB_DISTRIBUTOR is not set in /etc/default/grub which means
that the submenu comes out different than some of the desktop
Debian distro spins. This was able to resolve issue for me in testing.

msheiny committed:
Added STABLE config for default AWS kernel build

This was a pretty close copy of upstream AMI kernel config with grsec
settings sprinkled in,  in order to get testing working.
I'd like to strip out un-necessary components, such as sound card and
wireless drivers in another config. Hence the `full` designation on this config.

msheiny committed:
Potential fix for failing wait_for task

So I noticed that when trying to run against some of my EC2 instances
that the wait_for task is defaulting to `localhost` and bombs out since
I don't have a local SSH server listening. I re-ordered the host designator
so that inventory_hostname is tried first. I think this has to do with
everything running under the `local_action` module. I also stripped out the
ansible_ssh_host since that var is long-gone in ansible 2.0+

msheiny committed:
Revert "Potential fix for failing wait_for task"

This broke local packer builds for me but I wanted to keep the change-log
so decided to revert instead of rebasing and force-pushing.

conorsch commented on roles/install-grsec-kernel/tasks/grub_config.yml:
```diff
@@ -25,6 +25,17 @@
+  command: grep -q GRUB_DISTRIBUTOR /etc/default/grub
+  register: reg_grub_distributor_check
+  ignore_errors: yes
+  when: ansible_distribution == "Debian"
```
This'll need a `changed_when: false`, since it's a read-only task that will never change state.

conorsch commented:
Great commit comments, particularly f8eeab9. If I'm following that logic right, we should prepend `Debian ` to the grub submenu string if-and-only-if GRUB_DISTRIBUTOR is set in /etc/default/grub. Seems like a saner approach would be to read that value, but we can't rely on it being a string, as on some hosts it's enclosed in backticks, and defaults to "Debian".

Logic looks sound, I'll have to take this for a spin and try it out!

conorsch commented:
Rebased on top of latest master.

conorsch commented:
Functionally this is sound: I was able to build an AWS AMI running a grsecurity-patched kernel based off of this PR. Have a few opinions about style, so will add some revisions here prior to merge.

conorsch committed:
Sets read-only task to changed_when=false

Looking up values in the grub config will never changed state, so don't
report changed by default.

conorsch committed:
Makes grub submenu vars more readable

Updates the var names to follow the `_result` pragma for registered
variable, and gives a more descriptive name to the `grub_submenu_text`
var, describing what the value should contain and how it's used.

conorsch commented:
Cleaned up some stylistic nits. Taking a step back to evaluate whether the GRUB_DISTRIBUTOR check is the sanest way to determine the prefix. Looks like the submenu-text-deriving logic should be folded into the custom facts module and set there for all hosts. Will take a look and test on a few different backend providers to make sure it's working across the board.

msheiny commented:
>Taking a step back to evaluate whether the GRUB_DISTRIBUTOR check is the sanest way to determine the prefix.

Yay thats a good call. I made the detection decision very hastily to get things working. Havent tested across different distros besides debian locally, in AWS and DigitalOcean .

msheiny commented:
rebased ontop of `master`

conorsch committed:
Treats only Amazon EC2 instances specially in grub config

We can leverage Ansible host facts to determine whether we're likely
installing on an AWS EC2 image. If so, then provide a custom prefix to
the grub menu entry, otherwise the host will fail to come up after
reboot.

conorsch commented:
In the context of #109, we should get this in before splitting up the roles. Rebased the feature branch on latest master, and cleaned up the commits a bit. Confirmed that the changes do not affect installation on Ubuntu hosts in Vagrant VMs. Did not double-check against an AWS EC2 instance, but we can circle back on any problems once the install role has a dedicated repository.
